pyvips/vimage.py

```diff
@@ -998,33 +998,35 @@ def __getitem__(self, arg):
             last_two = rgb_image[1:]
             last_band = rgb_image[-1]
             middle_few = multiband[1:-2]
+            reversed  = multiband[::-1]
+            every_other = multiband[::2]
+            other_every_other = multiband[1::2]
+
+        In all cases, the semantics of slicing exactly match those of slicing 
+        `range(self.bands)`.
 
         """
 
+        band_range = range(self.bands)[arg] # raises IndexError for bad integer indices
```

Comment:
slicing a range object with an in-range integer yields an integer, while slicing it with a slice keeps it as a range object, so we let this do all the work of keeping track of start, stop, step, len, while also using the fact that bad integer slices will already raise an out-of-range `IndexError`.

Reply:
I didn't know about this feature. Neat!